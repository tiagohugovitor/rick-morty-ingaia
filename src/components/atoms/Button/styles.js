import styled from 'styled-components';

const StyledButton = styled.button`
  border: 2px solid ${(props) => props.theme.palette.white.main};
  box-sizing: border-box;
  border-radius: 8px;
  color: ${(props) => props.theme.palette.text.white};
  align-items: center;
  height: 100%;
  background: transparent;
  cursor: pointer;
  font-weight: ${(props) => props.theme.typography.weight[props.fontWeight]};;
  font-size: ${(props) => props.theme.typography.size[props.fontSize]};
  line-height: 22px;
  width: 100%;
`;

export {
  StyledButton,
};
