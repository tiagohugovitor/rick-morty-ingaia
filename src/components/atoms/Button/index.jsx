import React from 'react';
import PropTypes from 'prop-types';
import {StyledButton} from './styles';

const Button = ({
  label,
  id,
  name,
  disabled,
  onClick,
  fontSize,
  fontWeight,
}) => (
  <StyledButton
    aria-label={label}
    id={id}
    name={name}
    disabled={disabled}
    onClick={onClick}
    fontSize={fontSize}
    fontWeight={fontWeight}
  >
    {label}
  </StyledButton>
);

Button.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  onClick: PropTypes.func,
  fontSize: PropTypes.string,
  fontWeight: PropTypes.string,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  label: '',
  id: 'button-id',
  name: 'button-name',
  onClick: () => {},
  fontSize: 'small',
  fontWeight: 'normal',
  disabled: false,
};

export default Button;
