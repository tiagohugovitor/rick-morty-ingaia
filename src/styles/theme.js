import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  typography: {
    fontFamily: "nunito",
    size: {
      mini: '12px',
      tiny: '13px',
      small: '14px',
      mediumSmall: '16px',
      medium: '18px',
      large: '20px',
      huge: '24px',
      extra: '28px'
    },
    weight: {
      light: '300',
      normal: '400',
      bold: '700'
    }
  },
  palette: {
    primary: {
      main: "#CBD736",
      secondary: "#C9D636"
    },
    text: {
      primary: "#9F9F9F",
      secondary: "#8C8C8C",
      tertiary: "#CBD736",
      white: "#FFFFFF",
      black: "#000000",
      grey: "#D3D3D3"
    },
    warning: {
      main: "#FF0000"
    },
    white: {
      main: "#FFFFFF"
    },
    black: {
      main: "#000000"
    },
    background:{
      primary: "#1E1E1E",
      secondary: "#303030",
      tertiary: "#1B1B1B"
    },
    border: {
      primary: "#606060"
    }
  }
});

export default theme;