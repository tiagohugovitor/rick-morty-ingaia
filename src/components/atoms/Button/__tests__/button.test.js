import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import { ThemeProvider } from 'styled-components'
import theme from '../../../../styles/theme';
import Button from '../index';

configure({ adapter: new Adapter() });

describe('Render Button Component', () => {
  it('render correctly button component', () => {
    const ButtonComponent = renderer.create(
      <ThemeProvider theme={theme}>
        <Button />
      </ThemeProvider>
    ).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it('render correctly with default value of props', () => {
    const defaultProps = {};
    const ButtonComponent = mount(
      <ThemeProvider theme={theme}>
        <Button {...defaultProps} />
      </ThemeProvider>
    ).find(Button).first();
    expect((ButtonComponent).prop('label')).toEqual('');
    expect((ButtonComponent).prop('id')).toEqual('button-id');
    expect((ButtonComponent).prop('name')).toEqual('button-name');
    expect((ButtonComponent).prop('fontSize')).toEqual('small');
    expect((ButtonComponent).prop('fontWeight')).toEqual('normal');
    expect((ButtonComponent).prop('disabled')).toEqual(false);
  });

  it('render correctly with null values of props', () => {
    const defaultProps = {
      label: null,
      id: null,
      name: null,
      onClick: null,
      fontSize: null,
      fontWeight: null,
      disabled: null,
    };
    const ButtonComponent = mount(
      <ThemeProvider theme={theme}>
        <Button {...defaultProps} />
      </ThemeProvider>
    ).find(Button).first();
    expect((ButtonComponent).prop('label')).toEqual(null);
    expect((ButtonComponent).prop('id')).toEqual(null);
    expect((ButtonComponent).prop('name')).toEqual(null);
    expect((ButtonComponent).prop('fontSize')).toEqual(null);
    expect((ButtonComponent).prop('fontWeight')).toEqual(null);
    expect((ButtonComponent).prop('disabled')).toEqual(null);
  });

  it('render correctly with normal value of props', () => {
    const props = {
      label: "Search",
      id: "search-bar-button",
      name: "search-characters-button",
      disabled: false,
      fontSize: "mediumSmall",
      fontWeight: "bold"
    };
    const ButtonComponent = mount(
      <ThemeProvider theme={theme}>
        <Button {...props} />
      </ThemeProvider>
    ).find(Button).first();
    expect((ButtonComponent).prop('label')).toEqual('Search');
    expect((ButtonComponent).prop('id')).toEqual('search-bar-button');
    expect((ButtonComponent).prop('name')).toEqual('search-characters-button');
    expect((ButtonComponent).prop('fontSize')).toEqual('mediumSmall');
    expect((ButtonComponent).prop('fontWeight')).toEqual('bold');
    expect((ButtonComponent).prop('disabled')).toEqual(false);
  });

  it('check onClick callback', () => {
    const onClick = jest.fn();
    const props = {
      onClick: onClick
    };
    const ButtonComponent = mount(
      <ThemeProvider theme={theme}>
        <Button {...props} />
      </ThemeProvider>
    ).find(Button).first();
    ButtonComponent.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

});
