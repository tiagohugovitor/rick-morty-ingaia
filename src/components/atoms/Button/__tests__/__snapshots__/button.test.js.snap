// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Component render correctly button component 1`] = `
<button
  aria-label=""
  className="sc-bdvvaa fPfsNP"
  disabled={false}
  fontSize="small"
  fontWeight="normal"
  id="button-id"
  name="button-name"
  onClick={[Function]}
>
  
</button>
`;
